test-suite-java/src/test/java/io/micronaut/sourcegen/example/InnerTypesTest.java

```diff
@@ -0,0 +1,254 @@
+        PropertyDef.PropertyDefBuilder propertyBuilder = PropertyDef.builder("id").ofType(TypeDef.Primitive.INT);
+        recordBuilder.addProperty(propertyBuilder.build());
+
+        ClassDef.ClassDefBuilder classBuilder = getClassDefBuilderWith(recordBuilder.build());
+        assertAndClean(generateClass(classBuilder.build()), expectedString);
+    }
+
+    @Test
+    public void classInClass() throws IOException {
+        String expectedString = "class InnerClass {\n" +
+            "  class Inner {\n" +
+            "  }\n" +
+            "}\n";
+        ClassDef.ClassDefBuilder innerClassBuilder = ClassDef.builder("Inner");
+
+        ClassDef.ClassDefBuilder classBuilder = getClassDefBuilderWith(innerClassBuilder.build());
+        assertAndClean(generateClass(classBuilder.build()), expectedString);
+    }
+
+    @Test
+    public void interfaceInClass() throws IOException {
+        String expectedString = "class InterfaceClass {\n" +
+            "  interface Interface {\n" +
+            "  }\n" +
+            "}\n";
+        InterfaceDef.InterfaceDefBuilder interfaceBuilder = InterfaceDef.builder("Interface");
+
+        ClassDef.ClassDefBuilder classBuilder = getClassDefBuilderWith(interfaceBuilder.build());
+        assertAndClean(generateClass(classBuilder.build()), expectedString);
+    }
+
+    @Test
+    public void enumInRecord() throws IOException {
+        String expectedString = "record StatusRecord() {\n" +
+            "  enum Status {\n" +
+            "\n" +
+            "    SINGLE,\n" +
+            "    MARRIED\n" +
+            "  }\n" +
+            "}\n";
+        EnumDef.EnumDefBuilder enumBuilder = EnumDef.builder("Status");
+        enumBuilder.addEnumConstant("SINGLE").addEnumConstant("MARRIED");
+        EnumDef enumDef = enumBuilder.build();
+
+        RecordDef.RecordDefBuilder classBuilder = getRecordDefBuilderWith(enumDef);
+        assertAndClean(generateClass(classBuilder.build()), expectedString);
+    }
+
+    @Test
+    public void recordInRecord() throws IOException {
+        String expectedString = "record ExampleRecord() {\n" +
+            "  record Example(\n" +
+            "      int id\n" +
+            "  ) {\n" +
+            "  }\n" +
+            "}\n";
+        RecordDef.RecordDefBuilder recordBuilder = RecordDef.builder("Example");
+        PropertyDef.PropertyDefBuilder propertyBuilder = PropertyDef.builder("id").ofType(TypeDef.Primitive.INT);
+        recordBuilder.addProperty(propertyBuilder.build());
+
+        RecordDef.RecordDefBuilder classBuilder = getRecordDefBuilderWith(recordBuilder.build());
+        assertAndClean(generateClass(classBuilder.build()), expectedString);
+    }
+
+    @Test
+    public void classInRecord() throws IOException {
+        String expectedString = "record InnerRecord() {\n" +
+            "  class Inner {\n" +
+            "  }\n" +
+            "}\n";
+        ClassDef.ClassDefBuilder innerClassBuilder = ClassDef.builder("Inner");
```

Comment:
For verification, can you add another test with modifiers to class, e.g. static and private?